fhir-server/src/test/java/au/csiro/pathling/fhirpath/function/WhereFunctionTest.java

```diff
@@ -7,81 +7,92 @@
 package au.csiro.pathling.fhirpath.function;
 
 import static au.csiro.pathling.test.assertions.Assertions.assertThat;
+import static au.csiro.pathling.utilities.Preconditions.check;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
 
 import au.csiro.pathling.errors.InvalidUserInputError;
 import au.csiro.pathling.fhirpath.FhirPath;
+import au.csiro.pathling.fhirpath.ResourceDefinition;
 import au.csiro.pathling.fhirpath.ResourcePath;
 import au.csiro.pathling.fhirpath.element.ElementPath;
 import au.csiro.pathling.fhirpath.literal.BooleanLiteralPath;
 import au.csiro.pathling.fhirpath.parser.ParserContext;
-import au.csiro.pathling.io.ResourceReader;
 import au.csiro.pathling.test.builders.DatasetBuilder;
 import au.csiro.pathling.test.builders.ElementPathBuilder;
 import au.csiro.pathling.test.builders.ParserContextBuilder;
 import au.csiro.pathling.test.builders.ResourcePathBuilder;
+import au.csiro.pathling.test.helpers.FhirHelpers;
+import au.csiro.pathling.test.helpers.SparkHelpers;
+import au.csiro.pathling.test.helpers.SparkHelpers.IdAndValueColumns;
 import java.util.Arrays;
 import java.util.Collections;
-import org.apache.spark.sql.*;
+import java.util.Optional;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.functions;
 import org.apache.spark.sql.types.DataTypes;
 import org.apache.spark.sql.types.StructField;
 import org.apache.spark.sql.types.StructType;
+import org.hl7.fhir.r4.model.Encounter;
 import org.hl7.fhir.r4.model.Enumerations.FHIRDefinedType;
 import org.hl7.fhir.r4.model.Enumerations.ResourceType;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 
 /**
  * @author John Grimes
  */
 @Tag("UnitTest")
-@Disabled
 public class WhereFunctionTest {
 
+  // This test simulates the execution of the where function on the path
+  // `Patient.reverseResolve(Encounter.subject).where($this.status = 'in-progress')`.
   @Test
   public void whereOnResource() {
     final StructType encounterStruct = new StructType(new StructField[]{
         DataTypes.createStructField("id", DataTypes.StringType, true),
         DataTypes.createStructField("status", DataTypes.StringType, true)
     });
 
-    final Dataset<Row> dataset = new DatasetBuilder()
+    final Dataset<Row> inputDataset = new DatasetBuilder()
         .withIdColumn()
         .withStructTypeColumns(encounterStruct)
```

Comment:
There is no row with NULL value.

Reply:
See 9fb1a6e29c01238ea501bff04ba6a2cdd97bc3f2.